src/main/scala/mesosphere/marathon/core/base/CrashStrategy.scala

```diff
@@ -11,6 +11,6 @@ trait CrashStrategy {
 
 case object JvmExitsCrashStrategy extends CrashStrategy {
   override def crash(): Future[Done] = {
-    Runtime.getRuntime.asyncExit()(ExecutionContext.Implicits.global)
+    Runtime.getRuntime.asyncExit(RichRuntime.SuicideExitCode)(ExecutionContext.Implicits.global)
```

Comment:
Should we also start differentiating between suicide reasons? E.g. `def crash(reason: SuicideReason)` with e.g. `SuicideOnZKConnectionLoss` etc.?

Reply:
I was wondering the same. I'd say yes. However, it requires `def crash()` to take an argument.

Reply:
There are 5 usages of it in total and you can easily cover it with a default code.

Reply:
I'm with the zen-dog!

Reply:
+1 let's define several exit codes.

We have
* Failed to connect to Zookeeper
* Connection loss to Zookeeper
* Failed to initialize Plugins
* Lost leadership
* Leadership ended with failure
* Leadership ended gracefully
* Error in Mesos scheduler
* Uncaught exception

Am I missing any?